### DISCUSSION TITLE
Question about parameters

### DISCUSSION THREAD

**Discussion by jonschlinkert** [OAI/Arazzo-Specification#250] [General]
Hi, I'm just trying to learn the spec, but I might be confused about something. How do I define arguments on steps? I see `parameters`, which in every language I'm familiar with means "placeholders for arguments", while arguments are the actual values, but it looks like parameters are being used as arguments. Is there a different object that's used for args? FWIW I did search for this information first, but didn't find any discussions or answers about it. 

Thanks!

**Comment by frankkilcommins** [comment]
Hi @jonschlinkert,

Step parameters in most cases can contain _actual_ values or _placeholders_ for values, as well as specifying how the parameter should be mapped into a particular API operation (when applicable). In the _placeholder_ situation the value is normally expressed in the form of a runtime expression to be evaluated by tooling parsing/processing the Arazzo Description). The expression could specify that the value will be taken from provided workflow inputs, or perhaps from previous steps within a given workflow.

If you want to pass arguments into a workflow, then you would define the shape of the inputs using JSON Schema (see [Workflow inputs fixed field](https://spec.openapis.org/arazzo/v1.0.0.html#fixed-fields-2)).

If you have a concrete example, I can try to answer more explicitly for you.

**Reply by orubel** [reply]
It is important to note they cannot be associated with the incoming JWT ROLE in the token to provide RBAC and ABAC to be in compliance with OWASP (Photo courtesy of 'API's Explained (c))
![Screenshot from 2024-09-13 09-15-45](https://github.com/user-attachments/assets/7edd6a82-cc85-4140-a784-0871bf55bfb3)

To do this, you have to also provide a DIRECT way to associate ROLES with endpoints and check params sent per ROLE with the request as well as returned with the request (example provided via [spring-boot Beapi Plugin](https://github.com/Beapi-io/spring-boot-starter-beapi) with a [simple JSON schema](https://gist.github.com/orubel/6d529d39133e39b3f28d7cf1b9240236)
